File: Latihan2_2/python/Hardware.py
from Product import Product

# class Hardware adalah child dari class Product dan parent dari class Memory
class Hardware(Product):
    # atribut private
    __brand = ""
    __model = ""

    # constructor
    def __init__(self):
        self.__brand = ""
        self.__model = ""
    
    # set hardware brand
    def set_brand(self, brand):
        self.__brand = brand
    # get hardware brand
    def get_brand(self):
        return self.__brand
    
    # set hardware model
    def set_model(self, model):
        self.__model = model
    # get hardware model
    def get_model(self):
        return self.__model

File: Latihan2_2/cpp/Hardware.cpp
#include "Product.cpp"

// class Hardware adalah child dari class Product dan parent dari class Memory
class Hardware: public Product
{   
    // atribut private
    private:
        string brand;
        string model;
            
    public:
        // constructor 
        Hardware(string brand, string model) {
            this->brand = brand;
            this->model = model;
        }

        // constructor kosong
        Hardware() {
        }

        // set hardware brand
        void set_brand(string brand) {
            this->brand = brand;
        }
        // get hardware brand
         string get_brand() {
            return this->brand;
        }

        // set hardware model
        void set_model(string model) {
            this->model = model;
        }
        // get hardware model
        string get_model() {
            return this->model;
        }

        // destructor
        ~Hardware() {
        }
};